c10/util/Backtrace.cpp

```diff
@@ -282,25 +312,60 @@ std::string get_backtrace(
   // Toggles to true after the first skipped python frame.
   bool has_skipped_python_frames = false;
 
+  bool has_addr2line = true;
+  if (system("which addr2line > /dev/null 2>&1")) {
+    // addr2line command doesn't exist
+    has_addr2line = false;
```

Comment:
`get_backtrace` can get called a lot, so you are going to want to static this so that this subprocess call only happens once. It may also be a good idea to not test this by default, and gate by an envvar, to avoid unexpected surprises